### ISSUE TITLE
Implement Engine CRUD

### ISSUE THREAD

**Issue by mrzzy** [joeltio/bento-box#16]
Follow up on #15

Implement Engine Service CRUD to call corresponding ICS:
- [x] ApplySimulation
- [x] GetSimulation
- [x] ListSimulation
- [x] DropSimulation
- [x] Step Simulation
- [x] Get/Set Attributes.

**Comment by mrzzy** [comment]
@joeltio Which ICS functions should be called when I receive requests from Engine CRUD?

**Comment by joeltio** [comment]
There is currently no ICS function for you to call. However, I foresee that ApplySimulation will probably call something similar in ICS, but with ICS parameters like:
```c++
void applySimulation(ics::ComponentStore compStore, ics::IndexStore, indexStore, ...)
```

**Comment by mrzzy** [comment]
#27  Implements GRPC methods engine side without tests.

**Comment by mrzzy** [comment]
Implemented by PR #27
